File: src/main/resources/Calculator.rb
class Calculator
  include Java::fr.johannlecocq.testjava.scripting.interfaces.Calculator

  def add(i,j)
    return i+j
  end

  def mul(i,j)
    return i*j
  end

  def sub(i,j)
    return i-j
  end

  def div(i,j)
    return i/j
  end

end

File: src/main/resources/Calculator.js
var CalculatorType = Java.type("fr.johannlecocq.testjava.scripting.interfaces.Calculator");

var Calculator = Java.extend(CalculatorType, {
        add: function (i, j) {
            return i + j;
        },
        mul: function (i, j) {
            return i * j;
        },
        sub: function (i, j) {
            return i - j;
        },

        div: function (i, j) {
            return i / j;
        }
    }
);